docs/architecture/0e-plans.md

```diff
@@ -262,30 +264,44 @@ In an exposed header, the minimum work for 0ε is:
 * Ensure that exposed interfaces are not listed in the rendered Doxygen documentation.
 * Ensure that applications cannot inadvertently call private functions that are declared in exposed headers.
 
+Note that due to a lack of bandwidth, sample programs may keep using private interfaces.
+
 #### TF-PSA-Crypto header locations
 
-We distinguish between three categories of headers:
+We distinguish between four categories of headers:
 
 * Public headers define public, stable APIs.
-* Exposed headers define exposed interfaces, as well as private interfaces used by Mbed TLS. They need to be present when building Mbed TLS.
-* Private headers are only needed when building TF-PSA-Crypto itself.
+* Exposed headers define exposed interfaces, as well as private interfaces used by Mbed TLS. They need to be present when building Mbed TLS, but their content must clearly be excluded from the stable API.
+* Private headers do not define any exposed interfaces, but they are used by Mbed TLS for the time being. They need to be present when building Mbed TLS, but their content must clearly be excluded from the stable API.
+* Internal headers are only needed when building TF-PSA-Crypto itself. Their content must clearly be excluded from the stable API.
 
-The following table summarizes the characteristics of each category.
+The following table summarizes the characteristics of each category and their location in TF-PSA-Crypto 0ε.
 
-| Category | Location | In include path? |
+| Category | Location | Installed? | Doxygen? | Visible to Mbed TLS? |
 | -------- | -------- | ---------------- |
-| Public   | `include` | yes |
-| Exposed  | `include` | yes except for Doxygen |
-| Private  | `drivers/*/include` | only for crypto |
+| Public   | `include/mbedtls` | yes | yes | yes |
+| Exposed  | `drivers/*/include/mbedtls/private` | yes | no | yes |
+| Private  | `drivers/*/include/mbedtls/private` | yes for now | no | yes |
+| Internal | anything except `**/include/mbedtls/*.h` | no | no | no |
```

Comment:
Just to make this "anything except" more concrete, this will typically be `drivers/builtin/src/*.h` but can also be in other locations for other drivers.